tuna/worker_interface.py

```diff
@@ -336,21 +337,24 @@ def compose_fdb_entry(self, session, fin_json, fdb_obj):
     if 'time' in fdb_obj:
       fdb_entry.kernel_time = fdb_obj['time']
 
+    fdb_entry, _ = self.get_fdb_entry(session, solver)
+    fdb_entry.kernel_group = fdb_entry.id
+
     return fdb_entry
 
-  def compose_kernel_entry(self, fdb_obj, fdb_entry):
+  def compose_kernel_entry(self, session, fdb_obj, fdb_entry):
     """Compose a new Kernel Cache entry from fin input"""
     # Now we have the ID, lets add the binary cache objects
-    fdb_entry.blobs = []
     for kern_obj in fdb_obj['kernel_objects']:
       kernel_obj = self.dbt.kernel_cache()
-      kernel_obj.conv_find_db_key = fdb_entry.id
       kernel_obj.kernel_name = kern_obj['kernel_file']
       kernel_obj.kernel_args = kern_obj['comp_options']
       kernel_obj.kernel_blob = bytes(kern_obj['blob'], 'utf-8')
       kernel_obj.kernel_hash = kern_obj['md5_sum']
       kernel_obj.uncompressed_size = kern_obj['uncompressed_size']
-      fdb_entry.blobs.append(kernel_obj)
+
+      kernel_obj.kernel_group = fdb_entry.kernel_group
+      session.add(kernel_obj)
```

Comment:
needs a try catch and rollback in case of err

Reply:
Added a retry loop in a different branch. Will merge with that one.